opentelemetry-sdk/src/resource/builder.rs

```diff
@@ -0,0 +1,102 @@
+use std::time::Duration;
+
+use opentelemetry::KeyValue;
+
+use super::{Resource, ResourceDetector};
+
+/// Builder to allow easy composition of a Resource
+#[derive(Debug)]
+pub struct ResourceBuilder {
+    resource: Resource,
+}
+
+impl Default for ResourceBuilder {
+    /// Create ResourceBuilder with [Resource::default()].
+    fn default() -> Self {
+        ResourceBuilder {
+            resource: Resource::default(),
+        }
+    }
+}
+
+impl ResourceBuilder {
+    /// Create ResourceBuilder with [Resource::empty()].
+    pub fn new() -> Self {
+        ResourceBuilder {
+            resource: Resource::empty(),
+        }
+    }
+
+    /// Add a single [ResourceDetector] to your resource.
+    pub fn with_detector(self, detector: Box<dyn ResourceDetector>) -> Self {
+        self.with_detectors(vec![detector])
+    }
+
+    /// Add multiple [ResourceDetector] to your resource.
+    pub fn with_detectors(mut self, detectors: Vec<Box<dyn ResourceDetector>>) -> Self {
+        self.resource = self
+            .resource
+            .merge(&Resource::from_detectors(Duration::from_secs(0), detectors));
+        self
+    }
+
+    /// Add a [KeyValue] to the resource.
+    pub fn with_key_value(self, kv: KeyValue) -> Self {
+        self.with_key_values(vec![kv])
+    }
+
+    /// Add multiple [KeyValue]s to the resource.
+    pub fn with_key_values<T: IntoIterator<Item = KeyValue>>(mut self, kvs: T) -> Self {
+        self.resource = self.resource.merge(&Resource::new(kvs));
+        self
+    }
+
+    /// Create a [Resource] with the options provided to the [ResourceBuilder].
+    pub fn build(self) -> Resource {
```

Comment:
Can we implement `Into<Resource>` for `ResourceBuilder` to make it easier to use `ResourceBuilder` as `Resource` in parameters

Reply:
I did the From<ResourceBuilder> because clippy was complaining since it gives Into for free